webapp/security/schemas.py

```diff
@@ -1,33 +1,97 @@
-from marshmallow import fields, Schema
+import dateutil.parser
+from marshmallow import Schema, ValidationError
+from marshmallow.fields import DateTime, Dict, Float, List, Nested, Str
+from marshmallow.validate import Regexp
+
+
+from webapp.security.database import db_session
+from webapp.security.models import Release
+
+
+# Types
+# ===
+
+
+class ParsedDateTime(DateTime):
+    default_error_messages = {"parse_error": "dateutil cannot parse {input}."}
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        try:
+            date = dateutil.parser.parse(value)
+        except (OverflowError, ValueError):
+            raise self.make_error("parse_error", input=value)
+
+        return super()._deserialize(date.isoformat(), attr, data, **kwargs)
+
+
+# Schemas
+# ===
+
+
+# Notices
+# --
+def _validate_release_codenames(release_packages):
+    codenames = set(rel.codename for rel in db_session.query(Release).all())
+    unrecognised_codenames = set(release_packages.keys()) - codenames
+
+    if unrecognised_codenames:
+        raise ValidationError(
+            f"Unrecognised release codenames: {unrecognised_codenames}"
+        )
+
+
+class Package(Schema):
```

Comment:
There is another class named `Package` bellow.

Reply:
If we are going to do this I think we need to know if the package is a binary or a source and we might as well ask for them to send a `source_link` and `version_link` for packages that have them.

Reply:
I didn't see anywhere in the code that looked for binary packages - did I miss it?

Reply:
You didn't miss it. I was changin it on the PR we discussed yesterday to fix some package list differences with `usn.ubuntu.com`

Reply:
You can just update the class name, and I can fix the rest in a follow up PR